src/engine/Game.ts

```diff
@@ -0,0 +1,117 @@
+enum Results {
+	FIRST_PLAYER_WIN,
+	SECOND_PLAYER_WIN,
+	DRAW,
+	PENDING
+}
+
+export default class Game {
+	currentBoard: number[];
+	history: number[][];
+
+	constructor() {
+		this.history = [];
+		this.currentBoard = this.createEmptyBoard();
+	}
+
+	public getAvailableMoves(): number[] {
+		const availableMoves: number[] = [];
+
+		for ( let x = 0; x < this.currentBoard.length; x++ ) {
+			if ( this.currentBoard[ x ] === 0 ) {
+				availableMoves.push( x );
+			}
+		}
+
+		return availableMoves;
+	}
+
+	public getResult(): Results {
```

Comment:
We can make it a bit smarter and shorter. Consider creating a helper function that can return the result for 3 tiles.